api/v1alpha1/database_types.go

```diff
@@ -38,6 +40,14 @@ type DatabaseSpec struct {
 	// +optional
 	Datastreams *DatastreamsConfig `json:"datastreams,omitempty"`
 
+	// The state of the Database processes. Can be one of `Paused`, `Running` or `Frozen`.
+	// `Paused` means all the Storage Pods are being killed, but the Storage resource is persisted.
+	// `Frozen` means all the Pods are running, but the reconcile is effectively turned off.
+	// `Running` means the default state of the system, all Pods running.
+	// +kubebuilder:default:=Running
+	// +optional
+	Pause string `json:"pause,omitempty"`
```

Comment:
I think you underestimate the complexity that comes from introducing 2 orthogonal fields, and they are NOT completely orthogonal. If we do as you suggest, we will have 4 states: `(non-frozen, non-paused)`, `(frozen, non-paused)`, `(non-frozen, paused)` and `(frozen, paused)`. One of these states - (frozen, paused) does not even make any sense. If we introduce these 2 arguments, we will have to process 6 different transitions (number of edges in the graph with 4 vertices), and this will bring a lot of complexity into the codebase. We may additionally discuss this later in voice

Reply:
`Frozen` parameter does not correlate with state of Storage/Database and it's dependent objects. What do I mean by that? The main goal of this feature that we can manual edit StatefulSet/ConfigMap/Service and another dependent objects with sure that ydb-operator do not override our changes. This operation is needed for example during maintenance and migration actions. During this period, the YDB engineer assumes control of the Storage/Database (state and condition) under his responsibility. We can not stop StatefulSet (or Service for example) k8s controller, but client can use some special k8s tricks like `orphan=true` parameter to prevent cascade deletion or `kubectl replace` to change spec with deletion of object. In that case we can only guarantee that 1) ydb-operator reconcile disabled and does not sync CR and dependent objects 2) no one parameter can be changed  while Reconcile function is disabled and using webhook to verify that (we can edit CR resources only when state `Frozen` set to false)

In opposite `Pause` parameter correlate with state of Storage/Database and it's dependent objects. The main goal of this feature that we can do force stop processes without additional checks. This operation is needed for example during ad-hoc situations and cluster monstrous problems. Also we should change object's state to `Paused` whihc indicate that it's desired state when no one process is running. Additional function of this field that prevent deletion of Storage/Database object (we can delete CR resources only when state `Pause` set to false + for Storage we should additional check exist of dependent Database objects). To prevent data loss and inconsistency state we should implement finalizer to Pods with mandatory request CMS approve before killing process (please look at YDBOPS-6003).